### ISSUE TITLE
Large command output fails to post to API

### ISSUE THREAD

**Issue by cjshawMIT** [rtfd/readthedocs.org#1862]
Hi, I'm getting a 413 HTTP error on my build, after the `cat conf.py` step (I believe due to the first build step). When I build locally with Sphinx 1.3.2, I get a ton of warnings (96k+), but the build completes. I wonder if you have seen this error before, or if you think it is related to the number of warnings (a 413 is request entity too large)? Is there any way to suppress the warnings or force the build to continue?

```
Error
Client Error 413: https://readthedocs.org//api/v2/command/
```

**Comment by agjohnson** [comment]
I'd need to see the build that failed, can you paste a link to the project build that failed?

**Comment by cjshawMIT** [comment]
@agjohnson here is [one] (http://readthedocs.org/projects/dlkit-doc/builds/3557749/). I've been reducing the number of errors, and that does seem to be the culprit. I'm now down to 256, at it at least builds past that first step. Though I am getting a random "killed" error / build-failed when it tries to [do the `readthedocssinglehtmllocalmedia` build](http://readthedocs.org/projects/dlkit-doc/builds/3564861/)...(should I open a separate issue for that?). I'm not sure if this build failure is also related to the number of warnings I get, since I still have 256.

**Comment by agjohnson** [comment]
Yeah, my guess is the amount of text output we're trying to post to the api during the build process, not necessarily the number of build errors.

The second error looks like the process is being killed due to too much memory being consumed. This is a safety feature implemented to reduce abuse of our resources, we're still working out what this means for projects periodically using a large amount of resources. Do you have an idea of how much memory is normally required for your build process? You can measure this on unix systems with `time` (not the bash builtin time), ie: `/usr/bin/time -v make html` or something simliar.

**Comment by cjshawMIT** [comment]
@agjohnson thanks for looking into both of those. I don't to hijack this thread, so let me know if I should open a new issue, but wanted to follow up on your memory comment. For the `localhtmlfile` build that is failing, I used `gnu-time` (on a Mac) to get this output -- not sure how to interpret it to see how much memory it is using:

    Command being timed: "sphinx-build -T -b singlehtml -d ../build/doctrees-singlehtmllocalmedia -D language=en . ../build/localmedia"
	User time (seconds): 109.21
	System time (seconds): 1.54
	Percent of CPU this job got: 97%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:53.08
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 3763044352
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 284
	Minor (reclaiming a frame) page faults: 249268
	Voluntary context switches: 4628
	Involuntary context switches: 11227
	Swaps: 0
	File system inputs: 387
	File system outputs: 239
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0

If I divide maximum resident set size by 4 (per this [thread](http://stackoverflow.com/questions/10035232/maximum-resident-set-size-does-not-make-sense), a bug in gnu-time), it looks like 900 GB of memory? That does seem crazy large...what is the max that I can run with `readthedocs`, and any tips for cutting that down, from your experience?  -Thanks!

**Comment by agjohnson** [comment]
Hrm, not sure how to interpret that number either. Testing on BSD with `/usr/bin/time -lp`, I get 100M RSS with our own docs -- that sounds about right.

So it seems that if the RSS number above was the number of kB, but interpreted as the number of pages, the conversion that would have happened was `rss = "pages" * page size`. So to convert back to kB, divide by page size?

```
569049088 / 4096 / 1024 = 135.67 MB
```

:confused: 

That is a reasonable amount, I think it still stands that there is just a large amount of error output that has trouble posting to the API. We might be able to increase this limit, and if not, at least truncate output gracefully.

**Comment by cjshawMIT** [comment]
Hm...thanks for troubleshooting. I ran the same calculation on the first build step (which passes), and the number also came out at some oddly high value (300GB per my original calculation method), so perhaps there is another "divide by 1024" needed to make it memory used?

In the meantime, I've re-riggered our documentation to build without imports and autodoc, which seems to have helped a lot. Now everything builds on `readthedocs`, though I've got more errors / warnings to clean up.

Thanks for helping me troubleshoot and get back to a passing build!
